parser/hiveql/README.md

```diff
@@ -0,0 +1,46 @@
+# hiveql-parser
+
+This package is inspired by https://github.com/qzchenwl/hiveql-parser.
+
+## Development Environment
+
+To make sure that all developers are using the same version of JDK and Maven, we install these tools into a Docker image.
+
+```bash
+cd $PWD
+docker build -t sqlflow:hiveql .
+```
+
+## Build
+
+Then we can start a Docker container that runs the Docker image.
+
+```bash
+docker run --rm -it -v $HOME:/root -v $PWD:/work -w /work sqlflow:hiveql bash
+```
+
+Please be aware the `-v $HOME:/root` binds the `$HOME` directory on the host to the `/root`, the home directory, in the container, so that when we run Maven in the container, it saves the downloaded jars into `$HOME/.m2`.
+
+In the container, we can type the following command to build this package into a jar `target/hiveql-parser-1.0-SNAPSHOT.jar`.
+
+```bash
+$ mvn package
+```
+
+To build standalone jar, use:
+```bash
+$ mvn clean compile assembly:single
+```
+
+## Run
+
+```bash
+$ javar -jar /path/to/hiveql-parser.jar /path/to/your-code.sql
+```
+
+## Examples
+
+```bash
+$ java -jar hiveql-parser.jar <(echo "select count(*) as count, myfield from &0rz") 2>/dev/null
+[1,39]: line 1:39 cannot recognize input near '&' '0rz' '<EOF>' in join source
```

Comment:
After ran
```bash
java -jar target/hiveql-parser-1.0-SNAPSHOT-jar-with-dependencies.jar < (echo "select count(*) as count, myfield from &0rz") 2>/dev/null
```
I got result: `bash: syntax error near unexpected token `('`

Reply:
> After ran
> 
> ```shell
> java -jar target/hiveql-parser-1.0-SNAPSHOT-jar-with-dependencies.jar < (echo "select count(*) as count, myfield from &0rz") 2>/dev/null
> ```
> 
> I got result: `bash: syntax error near unexpected token `('`

My bad, `<(` cannot be separated by a space.

Reply:
I see, `<(` cannot be separated by a space.

Reply:
Will fix this by changing the command line into `-jar target/hiveql-parser-1.0-SNAPSHOT.jar` in the next PR. Thanks for the reminder!